core/src/smartcontracts/wasm.rs

```diff
@@ -992,56 +1290,38 @@ impl<'wrld> Runtime<state::validator::Migrate<'wrld>> {
     }
 }
 
-impl<'wrld> import::ExecuteOperations<state::validator::Migrate<'wrld>>
+impl<'wrld> ExecuteOperationsAsValidatorMutably<state::validator::Migrate<'wrld>>
     for Runtime<state::validator::Migrate<'wrld>>
 {
-    #[codec::wrap]
-    fn execute_query(
-        query: QueryBox,
-        state: &mut state::validator::Migrate<'wrld>,
-    ) -> Result<Value, ValidationFail> {
-        debug!(%query, "Executing as validator during migration");
+}
 
-        query
-            .execute(state.common_mut().wsv)
-            .map(|lazy_value| match lazy_value {
-                LazyValue::Value(value) => value,
-                LazyValue::Iter(iter) => Value::Vec(iter.collect()),
-            })
-            .map_err(Into::into)
+/// Fake implementation of [`import::QueryOperationsToValidate`].
+///
+/// This is needed because `migrate()` entrypoint exists in the same binary as
+/// `validate_*()` entrypoints.
+///
+/// # Panics
+///
+/// Panics with error message if called, because it should never be called from
+/// `migrate()` entrypoint.
+impl<'wrld> import::QueryOperationsToValidate<state::validator::Migrate<'wrld>>
+    for Runtime<state::validator::Migrate<'wrld>>
+{
+    #[codec::wrap]
+    fn query_transaction_to_validate(
+        _state: &state::validator::Migrate<'wrld>,
+    ) -> VersionedSignedTransaction {
+        panic!("Validator `migrate()` entrypoint should not query transaction to validate")
```

Comment:
Another place where "query" looks confusing (for me). "get transaction" feels more natural.